web/profiles/joinup/joinup.profile

```diff
@@ -382,8 +382,20 @@ function joinup_entity_view_alter(array &$build, EntityInterface $entity, Entity
   // Add the "collection_context" contextual links group on community content
   // and solutions.
   if (JoinupHelper::isSolution($entity) || CommunityContentHelper::isCommunityContent($entity)) {
+    // The contextual links need to vary per user roles and per user og roles.
+    // Core already takes care of varying by roles by applying the
+    // user.permissions cache context and applying the permission hash in the
+    // contextual links. We need to include the corresponding data deriving from
+    // the og role cache context.
```

Comment:
Also add here _why_ they are varying by both: a moderator might have the `administer shared content` permission site wide, and a facilitator might have the permission to share content in their groups.